### DISCUSSION TITLE
Error in generating the makefile

### DISCUSSION THREAD

**Discussion by R-0-n-Y** [PrincetonUniversity/athena#542] [General]
python configure.py -g -b --prob gr_torus --coord=kerr-schild --flux hlle -hdf5 --hdf5_path /home/simulate/hdf5-1.10.5/hdf5 -mpi
make clean 
make

While doing the make, I am having this error!
error: ‘H5Pset_fapl_mpio’ was not declared in this scope; did you mean ‘H5Pset_fapl_stdio’?
error: ‘H5Pset_dxpl_mpio’ was not declared in this scope; did you mean ‘H5Pset_fapl_stdio’?

Could you help me out?

**Comment by changgoo** [comment]
Your hdf5 library need to be compiled with the mpi option enabled to use it with `-mpi` option.

**Reply by R-0-n-Y** [reply]
I have used the -mpi option.

python configure.py -g -b --prob gr_torus --coord=kerr-schild --flux hlle -mpi -hdf5 --hdf5_path /home/sbdas/hdf5-1.10.5/hdf5

Still the error exists.

src/outputs/athena_hdf5.cpp: In member function ‘virtual void ATHDF5Output::WriteOutputFile(Mesh*, ParameterInput*, bool)’:
src/outputs/athena_hdf5.cpp:417:3: error: ‘H5Pset_fapl_mpio’ was not declared in this scope; did you mean ‘H5Pset_fapl_stdio’?
  417 |   H5Pset_fapl_mpio(property_list_file, MPI_COMM_WORLD, MPI_INFO_NULL);
      |   ^~~~~~~~~~~~~~~~
      |   H5Pset_fapl_stdio
src/outputs/athena_hdf5.cpp:674:3: error: ‘H5Pset_dxpl_mpio’ was not declared in this scope; did you mean ‘H5Pset_fapl_stdio’?
  674 |   H5Pset_dxpl_mpio(property_list, H5FD_MPIO_COLLECTIVE);
      |   ^~~~~~~~~~~~~~~~
      |   H5Pset_fapl_stdio

**Reply by changgoo** [reply]
I mean that your hdf5 library needs to be compiled (installed) with the mpi option to configure athena with hdf5 and mpi.

**Reply by R-0-n-Y** [reply]
Oh Okay! Let me reinstall hdf5. Thank you.

**Reply by R-0-n-Y** [reply]
After installing the hdf5 with mpi this message pops up

In file included from src/main.cpp:41:
src/outputs/outputs.hpp:20:10: fatal error: hdf5.h: No such file or directory
   20 | #include <hdf5.h>
      |          ^~~~~~~~
compilation terminated.
make: *** [Makefile:75: obj/main.o] Error 1

**Reply by c-white** [reply]
You probably need to use the `--hdf5_path` argument in the Athena configure script, in order to point to wherever the library files are. This is a question of the details of your hdf5 installation, not specific to Athena.

I suggest trying to get a hello world program, including the line `#include <hdf5.h>`, to compile and run on your machine. Then Athena should work the same way.

**Reply by c-white** [reply]
Addendum: If you are already using `--hdf5_path`, double check that the `include/` and `lib/` subdirectories are actually there.

**Reply by c-white** [reply]
A couple other suggestions: It might be worth checking what compiler is actually running. That is, does `mpicxx --version` actually point to a standard compiler like g++? (Weird compilers might require extra flags beyond what the configure script sets.) Also, just in case, make sure to `make clean` before attempting a new compilation.

If other programs can compile on your machine but Athena can't, it would be helpful to see the entire output, including which files successfully compiled before the failure.

**Reply by R-0-n-Y** [reply]
Thank you for the response.
